### DISCUSSION TITLE
Some of my videos are added without date, though the file does contain some

### DISCUSSION THREAD

**Discussion by Jayknightfr** [photoprism/photoprism#2229] [Q&A] [answered]
So i have many videos (MP4 format) that were added without any dates, here an exemple :  
![image](https://user-images.githubusercontent.com/19539311/162173907-5392d189-6ae5-4e55-bd69-11e473561c53.png)
So i wanted to use XnView  to batch edit all those videos that were taken at the same time, but the tool does show me the correct dates : 
![image](https://user-images.githubusercontent.com/19539311/162174399-8cc61fed-ba0e-4328-8cde-a5a248b91f54.png)
Which are also the data displayed by windows when i check the video file's properties : 
![image](https://user-images.githubusercontent.com/19539311/162174631-9f6fecc4-cfa3-4938-8fa0-6acc687084ce.png)

So what exactly am i doing wrong while scanning those videos?

**Comment by lastzero** [comment, marked as answer]
![Screenshot_20220407_121125](https://user-images.githubusercontent.com/301686/162176491-c1e65c69-dedc-4832-a969-5c0493d4d2bc.jpg)

Are you sure this is the right date? Or do you mean the file date (depends on when you added/copied it, not when it was really taken).

**Reply by Jayknightfr** [reply]
> ![Screenshot_20220407_121125](https://user-images.githubusercontent.com/301686/162176491-c1e65c69-dedc-4832-a969-5c0493d4d2bc.jpg)
> 
> Are you sure this is the right date? Or do you mean the file date (depends on when you added/copied it, not when it was really taken).

I'm sure it wasn't taken in 1969 :)
As the other dates are correct, i thought i red that the dates that were used are the creation date of the file? I don't know why the encoding date is set up like this, i guess i need to change that, but it's weird that this is the date that is used and not the file creation date?

**Reply by lastzero** [reply]
What other dates? Every file has a filesystem creation time, even an empty text file. It's not reliable to assume this is when a photo or video was created. I'm traveling right now, so can't look for the related docs to give you the URL. Should be described somewhere.

**Reply by Jayknightfr** [reply]
> What other dates? Every file has a filesystem creation time, even an empty text file. It's not reliable to assume this is when a photo or video was created. I'm traveling right now, so can't look for the related docs to give you the URL. Should be described somewhere.

Didn't you say that you'd use modification time when there was an unknown time?
https://github.com/photoprism/photoprism/discussions/1851

**Reply by lastzero** [reply]
Yes, for sorting in search results / albums. We don't show it as the actual creation date, which is unknown.

**Reply by Jayknightfr** [reply]
> Yes, for sorting in search results / albums. We don't show it as the actual creation date, which is unknown.

Ah, i see. Thanks for clarifying that. 

So which of the files' properties should i update for my videos? 
For this one in particular there are quite a lot of EXIF data related to dates : 

- Create Date
- Modify Date
- Track Create Date
- Track Modify Date
- Media  Create Date
- Media Modify Date

From the [exif.go](https://github.com/photoprism/photoprism/blob/b9ab3befe5d58a610022d0712676389804b43cd8/internal/meta/exif.go#L27)  code source i'd say it's "Create Date" ?

`var exifDateFields = []string{"DateTimeOriginal", "DateTimeDigitized", "CreateDate", "DateTime"}`

**Reply by Jayknightfr** [reply]
Anyway i used exiftool to change to createDate and the videos now have the correct date after indexing the files again

`sudo exiftool "-CreateDate<FileModifyDate" *.MP4
`
